### PR TITLE
#670 - Correcting IAM User Filters

### PR DESCRIPTION
**Pull request by joshuaroot**
* correcting issue causing AttributeError
* missing unittest added

### CODE REVIEW

**Comment by joshuaroot** [comment]
wip... adding boolean option for users without attached policy

File: c7n/resources/iam.py [hunk 1]
```diff
@@ -435,57 +435,53 @@ def match(self, info):
...
+            list(w.map(self.user_keys, resources))
 
-        matched = []
-        for r in resources:
-            for p in r['AccessKeys']:
-                if self.match(p):
-                    matched.append(r)
-                    break
-        return matched
+        if self.data.get('present', True):
```
Comment (kapilt) [inline c7n/resources/iam.py]: fwiw, these filters weren't intended as a boolean existence test, but a match on key attributes like CreationDate, which the previous impl tried to do via match().

Comment (joshuaroot) [inline c7n/resources/iam.py]: Ahhh, looks like I misunderstood. Looks like the original pulled only the attached_user_policies which doesn't provide additional attributes like 'CreateDate' only the PolicyName & ARN. Should be able to correct that easily.

**Comment by joshuaroot** [comment]
@kapilt -- so, test passes when i record the flight data but fails when i replay. All the requisite data seems to be stored within the placebo data so im not sure what's happening.

**Comment by kapilt** [comment]
@joshuaroot could you resolve conflict on this

**Pull request merged**